builtInFunctions/esdtDataStorage.go

```diff
@@ -421,14 +437,13 @@ func (e *esdtDataStorage) removeNFTMetadataFromSystemAccountIfNeeded(esdtTokenKe
 	if err != nil {
 		return err
 	}
-	if esdtData.Type != uint32(core.NonFungible) || tokenType != uint32(core.NonFungibleV2) {
+	if tokenType == esdtData.Type {
 		return nil
 	}
+	esdtData.Type = tokenType
```

Comment:
can we have a unit test for sft, metaesdt to check that the correct type is set?

Reply:
Added.